android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -723,74 +781,93 @@ Map<String, Integer> getCommandsMap() {
-        break;
-    }
-    super.receiveCommand(root, commandId, args);
+  public void receiveCommand(@NonNull RNCWebView view, String commandId, @Nullable ReadableArray args) {
+    view.ifHasInternalWebView(root -> {
+      switch (commandId) {
+        case "goBack":
+          root.goBack();
+          break;
+        case "goForward":
+          root.goForward();
+          break;
+        case "reload":
+          root.reload();
+          break;
+        case "stopLoading":
+          root.stopLoading();
+          break;
+        case "postMessage":
+          try {
+            InternalWebView reactWebView = (InternalWebView) root;
+            JSONObject eventInitDict = new JSONObject();
+            eventInitDict.put("data", args.getString(0));
+            reactWebView.evaluateJavascriptWithFallback("(function () {" +
+              "var event;" +
+              "var data = " + eventInitDict.toString() + ";" +
+              "try {" +
+              "event = new MessageEvent('message', data);" +
+              "} catch (e) {" +
+              "event = document.createEvent('MessageEvent');" +
+              "event.initMessageEvent('message', true, true, data.data, data.origin, data.lastEventId, data.source);" +
+              "}" +
+              "document.dispatchEvent(event);" +
+              "})();");
+          } catch (JSONException e) {
+            throw new RuntimeException(e);
+          }
+          break;
+        case "injectJavaScript":
+          InternalWebView reactWebView = (InternalWebView) root;
+          reactWebView.evaluateJavascriptWithFallback(args.getString(0));
+          break;
+        case "loadUrl":
+          if (args == null) {
+            throw new RuntimeException("Arguments for loading an url are null!");
+          }
+          ((InternalWebView) root).progressChangedFilter.setWaitingForCommandLoadUrl(false);
+          root.loadUrl(args.getString(0));
+          break;
+        case "requestFocus":
+          root.requestFocus();
+          break;
+        case "clearFormData":
+          root.clearFormData();
+          break;
+        case "clearCache":
+          boolean includeDiskFiles = args != null && args.getBoolean(0);
+          root.clearCache(includeDiskFiles);
+          break;
+        case "clearHistory":
+          root.clearHistory();
+          break;
+        case "release":
+          // no-op for now
+          break;
+      }
+    });
+    super.receiveCommand(view, commandId, args);
   }
 
   @Override
-  public void onDropViewInstance(WebView webView) {
-    super.onDropViewInstance(webView);
-    ((ThemedReactContext) webView.getContext()).removeLifecycleEventListener((RNCWebView) webView);
-    ((RNCWebView) webView).cleanupCallbacksAndDestroy();
-    mWebChromeClient = null;
+  public void onDropViewInstance(RNCWebView view) {
+    super.onDropViewInstance(view);
+
+    // The internal webview can be null since the view may have been already reattached
```

Comment:
Again, by "the view may have been already reattached," I assume you mean that the former internal WebView of this `view` may have been moved to a different `RNCWebView` container?

Reply:
Also, doesn't `ifHasInternalWebView` already handle the possibility of the internal WebView being null?